build/gen_stub.php

```diff
@@ -1480,23 +1483,380 @@ private function getFlagsAsArginfoString(): string
+            return $parametersRefSec;
+        } else {
+            $parametersPara = $doc->createElement('para');
+            $parametersRefSec->appendChild($parametersPara);
+
+            $parametersPara->appendChild(new DOMText("\n   "));
+            $parametersList = $doc->createElement('variablelist');
+            $parametersPara->appendChild($parametersList);
+
+            /*
+            <varlistentry>
+             <term><parameter>name</parameter></term>
+             <listitem>
+              <para>
+               Description.
+              </para>
+             </listitem>
+            </varlistentry>
+            */
+            foreach ($this->args as $arg) {
+                $parameter = $doc->createElement('parameter', $arg->name);
+                $parameterTerm = $doc->createElement('term');
+                $parameterTerm->appendChild($parameter);
 
-        return $doc->saveXML();
+                $parameterEntry = $doc->createElement('varlistentry');
+                $parameterEntry->appendChild(new DOMText("\n     "));
+                $parameterEntry->appendChild($parameterTerm);
+                $parameterEntry->appendChild(new DOMText("\n     "));
+
+                $listItemPara = $doc->createElement('para');
+                $listItemPara->appendChild(new DOMText("\n       "));
+                $listItemPara->appendChild(new DOMText("Description."));
+                $listItemPara->appendChild(new DOMText("\n      "));
+
+                $parameterEntryListItem = $doc->createElement('listitem');
+                $parameterEntryListItem->appendChild(new DOMText("\n      "));
+                $parameterEntryListItem->appendChild($listItemPara);
+                $parameterEntryListItem->appendChild(new DOMText("\n     "));
+
+                $parameterEntry->appendChild($parameterEntryListItem);
+                $parameterEntry->appendChild(new DOMText("\n    "));
+
+                $parametersList->appendChild(new DOMText("\n    "));
+                $parametersList->appendChild($parameterEntry);
+            }
+            $parametersList->appendChild(new DOMText("\n   "));
+        }
+        $parametersPara->appendChild(new DOMText("\n  "));
+        $parametersRefSec->appendChild(new DOMText("\n "));
+        return $parametersRefSec;
+    }
+
+    private function getReturnValueSection(DOMDocument $doc): DOMElement {
+        $returnRefSec = $this->generateRefSect1($doc, 'returnvalues');
+
+        $returnDescriptionPara = $doc->createElement('para');
+        $returnDescriptionPara->appendChild(new DOMText("\n   "));
+
+        $returnType = $this->return->type;
+        if ($returnType === null) {
+            $returnDescriptionPara->appendChild(new DOMText("Description"));
+        } else if (count($returnType->types) === 1) {
+            $type = $returnType->types[0];
+            $name = $type->name;
+            $descriptionNode = match ($name) {
```

Comment:
Unfortunately PHP 7.4 compatibility is required

Reply:
Can't we bump this for 8.4?

Reply:
No, unfortunately we cannot, since gen_stub runs during compilation via the PHP version which is on path. Since it's part of the build infrastructure, we have to carefully respect BC. Maybe we can require PHP 8.0 in a few years, or when PHP 9.0 comes out.